src/Tokenizer.sol

```diff
@@ -176,6 +184,11 @@ contract Tokenizer is UUPSUpgradeable, OwnableUpgradeable, IControlIPTs {
 
     /// @dev this will be called by IPTs. Right now the controller is the IPNFT's current owner, it can be a Governor in the future.
     function controllerOf(uint256 ipnftId) public view override returns (address) {
+        //todo: check whether this is safe (or if I can trick myself to be the controller somehow)
+        //reservations are deleted upon mints, so this imo should be good
+        if (ipnft.reservations(ipnftId) != address(0)) {
```

Comment:
is this correct:
`return ipnft.reservations(ipnftId) || ipnft.ownerOf(ipnftId)`

if the ipnft is minted it will return `ipnft.ownerOf(ipnftId)` as `ipnft.reservations(ipnftId)` would be `address(0)` abd vice versa.

Reply:
I would never assume by defaul whether a language besides Javascript would automatically cast an (address(0)) to a boolean (true). That code is pretty concise imo. If there's a reservation it returns its initiator, otherwise the holder.

Reply:
yes yes the code is clear, I just wanted to know whether the other optiion is also correct or not